sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -256,25 +187,22 @@ export function createReceiver(
     lastEnqueuedEventProperties: {},
     isClosed: false,
     close: async () => {
-      if (!receiver) {
-        return;
-      }
```

Comment:
There're cases where the underlying link is closed but the EH receiver is still open, this change makes sure to close the receiver if the link is already closed.